src/App.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div id="oc-file-picker" class="uk-height-1-1">
+    <div
+      v-if="state === 'loading'"
+      class="uk-height-1-1 uk-width-1-1 uk-flex uk-flex-middle uk-flex-center oc-border"
+    >
+      <oc-spinner aria-label="Loading ownCloud file picker" />
+    </div>
+    <login v-if="state === 'unauthorized'" key="login-form" @login="authenticate" />
+    <file-picker
+      v-if="state === 'authorized'"
+      key="file-picker"
+      class="uk-height-1-1"
+      :variation="variation"
+      @selectResources="selectResources"
+    />
+  </div>
+</template>
+
+<script>
+import Vue from 'vue'
+import sdk from 'owncloud-sdk'
+import DesignSystem from 'owncloud-design-system'
+import initVueAuthenticate from './services/auth'
+import FilePicker from './components/FilePicker.vue'
+import Login from './components/Login.vue'
+
+// Init sdk and design system
+Vue.prototype.$client = new sdk()
+Vue.use(DesignSystem)
+
+export default {
+  name: 'App',
+
+  components: {
+    FilePicker,
+    Login
+  },
+
+  props: {
+    variation: {
+      type: String,
+      required: true,
+      validator: value => value === 'resource' || value === 'location'
+    }
+  },
+
+  data: () => ({
+    authInstance: null,
+    bearerToken: '',
+    state: 'loading',
+    config: null
+  }),
+
+  created() {
+    this.checkAuthorization()
+  },
+
+  beforeDestroy() {
+    this.authInstance.mgr.events.removeUserLoaded()
+  },
+
+  methods: {
+    init() {
+      this.bearerToken = this.authInstance.getToken()
+
+      // Init owncloud-sdk
+      this.$client.init({
+        baseUrl: this.config.server,
+        auth: {
+          bearer: this.bearerToken
+        },
+        headers: {
+          'X-Requested-With': 'XMLHttpRequest'
+        }
+      })
+
+      this.state = 'authorized'
+
+      return
+    },
+
+    async checkAuthorization() {
+      let config = await fetch(window.location.origin + '/file-picker-config.json')
+
+      this.config = await config.json()
+      this.authInstance = initVueAuthenticate(this.config)
+
+      if (this.authInstance.isAuthenticated()) {
+        return this.init()
+      }
+
+      this.state = 'unauthorized'
```

Comment:
maybe move these two statements into an "authorize" method, just to make understanding easier

I just got confused why the above would call `this.init()` and skip the below code and whether it would break anything

Reply:
Done